crypto/crypto.go

```diff
@@ -37,3 +37,22 @@ func GenerateOtp(digits int) (string, error) {
 	otp := fmt.Sprintf(expr, val.String())
 	return otp, nil
 }
+
+// GenerateOtpFromCharset generates a random n-length otp from a charset
+func GenerateOtpFromCharset(length int, charset string) (string, error) {
+	b := make([]byte, length)
+	for i := range b {
+		val, err := rand.Int(rand.Reader, big.NewInt(int64(len(charset))))
+		if err != nil {
+			return "", errors.WithMessage(err, "Error generating otp from charset")
+		}
+		b[i] = charset[val.Int64()]
+	}
+	return string(b), nil
+}
+
+// GenerateEmailOtp generates a random n-length alphanumeric otp
+func GenerateEmailOtp(length int) (string, error) {
+	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
```

Comment:
I think it's ok but are there any situations where upper case chars could get squashed?